Source/JavaScriptCore/jit/JIT.h

```diff
@@ -668,11 +674,25 @@ namespace JSC {
         static MacroAssemblerCodeRef<JITThunkPtrTag> slow_op_get_by_val_callSlowOperationThenCheckExceptionGenerator(VM&);
         static MacroAssemblerCodeRef<JITThunkPtrTag> slow_op_get_private_name_callSlowOperationThenCheckExceptionGenerator(VM&);
         static MacroAssemblerCodeRef<JITThunkPtrTag> slow_op_get_from_scopeGenerator(VM&);
+        template<typename Bytecode>
+        ALWAYS_INLINE static MacroAssemblerCodeRef<JITThunkPtrTag> slow_op_get_from_scopeGenerator_helper(VM&, OperationFunction);
         static MacroAssemblerCodeRef<JITThunkPtrTag> slow_op_resolve_scopeGenerator(VM&);
+        template<typename Bytecode>
+        ALWAYS_INLINE static MacroAssemblerCodeRef<JITThunkPtrTag> slow_op_resolve_scopeGenerator_helper(VM&, OperationFunction);
         template <ResolveType>
         static MacroAssemblerCodeRef<JITThunkPtrTag> generateOpGetFromScopeThunk(VM&);
+        template <ResolveType, typename Bytecode>
+        ALWAYS_INLINE static MacroAssemblerCodeRef<JITThunkPtrTag> generateOpGetFromScopeThunkHelper(VM&, ThunkGenerator);
         template <ResolveType>
         static MacroAssemblerCodeRef<JITThunkPtrTag> generateOpResolveScopeThunk(VM&);
+        template <ResolveType, typename Bytecode>
+        ALWAYS_INLINE static MacroAssemblerCodeRef<JITThunkPtrTag> generateOpResolveScopeThunkHelper(VM&, ThunkGenerator);
+        static MacroAssemblerCodeRef<JITThunkPtrTag> slow_op_rgs_resolve_scopeGenerator(VM&);
+        static MacroAssemblerCodeRef<JITThunkPtrTag> slow_op_rgs_get_from_scopeGenerator(VM&);
+        template <ResolveType>
+        static MacroAssemblerCodeRef<JITThunkPtrTag> generateOpRGSResolveScopeThunk(VM&);
+        template <ResolveType>
+        static MacroAssemblerCodeRef<JITThunkPtrTag> generateOpRGSGetFromScopeThunk(VM&);
```

Comment:
nit: Let's not abbreviate this name this way. Write out the full thing

Reply:
Updated